packages/extension/src/components/pages/NFTViewer/NFTViewer.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, useEffect, useState } from 'react';
+
+import { AccountNFTokenResponse } from '@gemwallet/constants';
+
+import { useLedger } from '../../../contexts';
+import { NFTListing } from '../../organisms';
+import { PageWithHeader } from '../../templates';
+
+export const MAX_FETCHED_NFTS = 20;
+
+const initalState = {
+  account_nfts: [],
+  marker: null,
+  loading: false
+};
+
+interface NFTsProps extends AccountNFTokenResponse {
+  loading: boolean;
+}
+
+export const NFTViewer: FC = () => {
+  const { getNFTs } = useLedger();
+
+  const [NFTs, setNFTs] = useState<NFTsProps>(initalState);
+
+  const fetchNFTs = async () => {
```

Comment:
Instead of redefining this at every-lunch, we could either wrap it in a useCallback, or you could use it only in the useEffect.
You can find an example of async use here: packages/extension/src/contexts/NetworkContext/NetworkContext.tsx:47